src/azul/service/responseobjects/storage_service.py

```diff
@@ -133,6 +134,11 @@ def __complete(self):
             self.__abort()
             raise MultipartUploadError(self.bucket_name, self.object_key) from exception
 
+        if self.content_disposition:
+            s3_object = self.mp_upload.Object()
+            s3_object.copy_from(CopySource={'Bucket': self.bucket_name, 'Key': self.object_key},
```

Comment:
Apologies, I had forgotten that one can't update metadata on an object. Unless you can show me that no actual copy is made (I found no evidence online that S3 is optimizing this case), this is a show-stopper because it's too expensive for a convenience feature like this. 

Instead, can we create a shadow object (same key but append `.name` or something) that stores the name and then use it when creating the signed URL to the actual manifest?